src/crate/client/http.py

```diff
@@ -258,6 +283,30 @@ def _create_sql_payload(stmt, args, bulk_args):
     return json.dumps(data, cls=CrateJsonEncoder)
 
 
+def _get_socket_opts(keepalive=True,
+                     tcp_keepidle=None,
+                     tcp_keepintvl=None,
+                     tcp_keepcnt=None):
+    """
+    Return an optional list of socket options for urllib3's HTTPConnection
+    constructor.
+    """
+    if not keepalive:
+        return None
+
+    # Let's always use TCP keepalive.
+    opts = [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)]
+
+    if hasattr(socket, 'TCP_KEEPIDLE') and tcp_keepidle is not None:
```

Comment:
```suggestion
    # hasattr check because some of the options depend on system capabilities
    # see https://docs.python.org/3/library/socket.html?highlight=socket#socket.SOMAXCONN
    if hasattr(socket, 'TCP_KEEPIDLE') and tcp_keepidle is not None:
```